src/FileProvider/Structs.cs

```diff
@@ -0,0 +1,18 @@
+#if MONOMAC
+using System;
+using System.Runtime.InteropServices;
+using ObjCRuntime;
+
+namespace FileProvider {
+
+	[NoiOS, NoMacCatalyst, Mac (12,0)]
+	[StructLayout (LayoutKind.Sequential)]
+	public struct NSFileProviderTypeAndCreator
```

Comment:
I was looking for a better way of binding this since they are OSType but I was not sure which values to fit in here :/

Reply:
OSType is apparently a [FourCharCode  (FourCC)](https://developer.apple.com/documentation/kernel/ostype), so we could add methods that return the FourCC value:

```csharp
public string GetTypeAsFourCC ()
{
    return Runtime.ToFourCCString (Type);
}
public string GetCreatorAsFourCC()
{
    return Runtime.ToFourCCString (Creator);
}
```

The Runtime.ToFourCCString doesn't exist yet, but soon will: https://github.com/xamarin/xamarin-macios/pull/12305

Reply:
Lets wait for the PR and add the methods.